### DISCUSSION TITLE
Query - Conversion error on binding null value to numeric type parameter

### DISCUSSION THREAD

**Discussion by Bitello** [Budibase/budibase#9532] [Help] [answered]
Hi!

I have a problem with a Query with a stored procedure from a legacy system.

I'm using MS SQL Server, the procedure has some input parameters, all with the null option.
![image](https://user-images.githubusercontent.com/22456825/216624351-98769298-4f5d-44f2-ac2d-a23e3fd6f1f0.png)

When the procedure's parameter is of type char or date/time, I don't need to assign values to the Binding Pattern attribute (that is, it is understood as a Null value).
The same does not occur when the parameter is of numeric data type (all numeric types), resulting in a type conversion error.
![image](https://user-images.githubusercontent.com/22456825/216624565-08ff4520-c912-4d1e-a6e9-a1b592acdd20.png)

If I assign to Bindings whose parameters are of numerical type default value 0 (numerical zero), it works! 
But that's not a solution since they are "optional parameters".

Even if I change the type of the parameter to varchar in the procedure, when I try to convert it to the correct numerical type, Budibase reports an error (running the procedure in Management Studio works correctly).
![image](https://user-images.githubusercontent.com/22456825/216624726-c7c669a8-cc17-4060-af13-e24dfee8a9a9.png)

Any suggestions?

Grateful

**Comment by Bitello** [comment]
Another curiosity is that when this example procedure, which just returns a number and its data type,

![image](https://user-images.githubusercontent.com/22456825/216660875-7dcd1afa-b003-4cdd-8f31-8348d9913de1.png)

when executed by Budibase, setting the value 0 (zero) as Default, the JSON returns "0" (zero as a string) and the data type as numeric,
![image](https://user-images.githubusercontent.com/22456825/216660935-e644406f-4ed6-4a17-be79-bf1215e17c1c.png)

if I set a default value other than 0, such as 19681, JSON returns a numerical value, as its data type.
![image](https://user-images.githubusercontent.com/22456825/216661031-87ecab78-8f50-4ef0-a082-087db9a4ca19.png)

**Reply by Bitello** [reply]
I understand why it is generating the error.

Budibase types all bindings as being of nvarchar data type while expected parameters in procedures are "strongly" typed.

exec sp_executesql @statement=N'exec cpProtocoloListaX_Bitello
@p0, 
@p1',@params=N'@p0 **nvarchar(1)**, @p1 **nvarchar(3)**',@p0=N'',@p1=N'152'


In the company where I work, there is an internal tool system developed at the end of the 1990s, client server, windows form, and (which was the directive, at the time, recommended by Microsoft) with all database queries (SQL Server ) being performed through stored procedures (there are more than 3,000 procedures).

I imagined using Budibase as a "quick way" to migrate this legacy system to the WEB, but having to change hundreds, maybe thousands of procedures, is not feasible.

When I changed the call created by Budibase to consume the SQL procedure by changing the nvarchar parameters to the expected data type and running it through Management Studio, there was no data type conversion error.

**I would like to suggest the possibility of typing the data type of Binding.**

The potential of Budibase goes beyond the "new", being a tool that helps thousands of legacy systems that exist around the world to migrate to the future is something that Budibase could explore.

**Comment by melohagan** [comment, marked as answer]
Thanks for the detail @Bitello 

I've raised your issue: https://github.com/Budibase/budibase/issues/9683
